apps/core/faviconmanager.cpp

```diff
@@ -0,0 +1,218 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <QDebug>
+#include <QFile>
+#include <QTextStream>
+#include <QJsonObject>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QUrl>
+#include <memory>
+
+#include "browserpaths.h"
+#include "datafetcher.h"
+
+#include "faviconmanager.h"
+
+FaviconManager::FaviconManager(QObject *parent)
+    : QObject(parent)
+    , m_faviconSets()
+{
+}
+
+FaviconManager *FaviconManager::instance()
+{
+    static FaviconManager *singleton;
+    if (!singleton) {
+        singleton = new FaviconManager();
+    }
+
+    return singleton;
+}
+
+static QString sanitiseHostname(const QString &hostname)
+{
+    QUrl url(hostname);
+    // Maybe port should be included too
+    return QString("%1://%2").arg(url.scheme()).arg(url.host());
+}
+
+void FaviconManager::save(const QString &type)
+{
+    if (!m_faviconSets.contains(type)) {
+        qDebug() << "No" << type << "favicons loaded, skipping save";
```

Comment:
Sorry that I missed this earlier.

Let's add something like:
```
Q_LOGGING_CATEGORY(lcFavoritesLog, "org.sailfishos.browser.favorites", QtWarningMsg)
```
to the logging.cpp and use lcFavoritesLog here (qCDebug, qCWarning)

Reply:
Thanks, I should have done this from the start. Should be there now.